door-open-service/node-backend/src/StatsDatabase.js

```diff
@@ -45,41 +49,52 @@ DoorOpens.sync();
 class StatsDatabase {
 
   static registerDoorOpen(user) {
-    DoorOpens.create(user);
+    DoorOpens.create({ user: user.name, email: user.email, type: "door" });
   }
 
-  static registerGarageOpen(/*user*/) {
-    // TODO: register garage openings
+  static registerGarageOpen(user) {
+    DoorOpens.create({ user: user.name, email: user.email, type: "garage" });
   }
 
   static getStats() {
-    const queries = {
-      count_per_dayQ: sequelize.query(
-        "SELECT case cast(strftime('%w', timestamp) as integer) " +
-        "when 0 then 'Sun' " +
-        "when 1 then 'Mon' " +
-        "when 2 then 'Tue' " +
-        "when 3 then 'Wed' " +
-        "when 4 then 'Thu' " +
-        "when 5 then 'Fri' " +
-        "else 'Sat' end as dayofweek, COUNT(timestamp) as count FROM door_opens GROUP BY dayofweek", {
-          type: sequelize.QueryTypes.SELECT
-        }),
-      timestampQ: DoorOpens.findOne({
-        order: "timestamp ASC",
-        limit: 1,
-        attributes: ["timestamp"]
-      })
+    const queries = type => {
+      return {
+        count_per_dayQ: sequelize.query(
+          "SELECT case cast(strftime('%w', timestamp) as integer) " +
+            "when 0 then 'Sun' " +
+            "when 1 then 'Mon' " +
+            "when 2 then 'Tue' " +
+            "when 3 then 'Wed' " +
+            "when 4 then 'Thu' " +
+            "when 5 then 'Fri' " +
+            "else 'Sat' end as dayofweek, COUNT(timestamp) as count FROM door_opens " +
+            `WHERE type = '${type}' GROUP BY dayofweek`, {
+              type: sequelize.QueryTypes.SELECT
+            }),
+        timestampQ: DoorOpens.findOne({
+          order: "timestamp ASC",
+          where: { type: type },
```

Comment:
`{ type }` is a shorthand for these common cases where the name of the variable is the same as the field name.